static/src/javascripts/projects/commercial/modules/dfp/define-slot.js

```diff
@@ -139,9 +141,21 @@ const defineSlot = (adSlotNode: Element, sizes: Object): Object => {
             loadedResolve = resolve;
         });
 
-        const iasDataCallback = () => {
-            iasPET.setTargetingForGPT();
+        const iasDataCallback = targetingJSON => {
             clearTimeout(timeoutId);
+
+            /*  There is a name-clash with the `fr` targeting returned by IAS
+                and the `fr` paramater we already use for frequency. Therefore
+                we apply the targeting to the slot ourselves and rename the IAS
+                fr parameter to `fra` (given that, here, it relates to fraud).
+            */
+            const targeting = JSON.parse(targetingJSON);
```

Comment:
Is there a chance this `parse` call could fail? If so, it is possible that `iasDataPromise` will never resolve. It might be worth wrapping this section of code in a `try... catch` block just in case. What do you think?

Reply:
>Is there a chance this `parse` could fail?
Famous last words but: I don't think so. This function is only ever called by the IAS integration when it has JSON targeting to be applied. If the parsing _were_ to somehow fail, then the timeout promise should fire and resolve the `iasDataPromise` itself.